### ISSUE TITLE
Associate entry with toplevel comment and module comment

### ISSUE THREAD

**Issue by vch9** [ocaml/merlin#1421]
Hello, this is more likely a request for help than an actual issue.

I am working with @rgrinberg fork vendored (similarly to `ocaml-lsp`), and I'm trying to work with
documented/undocumented entries in a OCaml program.

Using Merlin I can for now detect entries in a program (types, values, modules etc.) and with [odoc](https://github.com/ocaml/merlin/blob/master/src/analysis/ocamldoc.ml) determine whether they are documented or not. It is currently working fine except for two specific cases:

1. Toplevel documentation with a license header:
```ocaml
(*******************************)
(* This is a wonderful license *)
(*******************************)

(** Toplevel documentation *)

type t = ...
...
```

2. Module documentation inside module:
```ocaml
module Foo = struct
  (** Foo documentation *)

end
```

In the first Merlin does not seem to be capable to find the toplevel documentation. However, in the second case `merlin-document` on `Foo` succeeds to find the documentation. But, I was not able to manually reproduce `merlin-document` using [odoc](https://github.com/ocaml/merlin/blob/master/src/analysis/ocamldoc.ml).

I created a "minimal" [repository](https://github.com/vch9/merlin-analysis-module) with my actual use of the `merlin.analysis`.

There are the two only corner-cases I've found so far. Would you have any ideas on why they can not find the associated comments ?

**Comment by voodoos** [comment]
Hello @vch9, what your are trying to achieve is not entirely clear to me. Are you using odoc for comparison with Merlin results or do you plan to mix the two ?

Also I think there is indeed an issue with Merlin but it is not linked to the licence header: can you try removing the space between the documentation string and the type declaration ?
```
(** Toplevel documentation *)
type t = ...
```

In my attempts Merlin find the doc only when there is no space, can you confirm that ?

I am not (yet) aware of the exact mechanics of Merlin's document command, @trefis , is that a bug or a feature ?

**Comment by voodoos** [comment]
In fact this behaviour respects the [`ocamldoc` specification](https://ocaml.org/manual/ocamldoc.html#s%3Aocamldoc-comments):

> ## Comments in mli files:
> A special comment before an element is associated to this element if ‍:
> 
> - There is no blank line or another special comment between the special comment and the element. However, a regular comment can occur between the special comment and the element.
> - The special comment is not already associated to the previous element.
> - The special comment is not the first one of a toplevel module. 
> 
> A special comment after an element is associated to this element if there is no blank line or comment between the special comment and the element.
> 
> There are two exceptions: for constructors and record fields in type definitions, the associated comment can only be placed after the constructor or field definition, without blank lines or other comments between them. The special comment for a constructor with another constructor following must be placed before the ’|’ character separating the two constructors.

So if I am not mistaken there is no issue here.

**Comment by vch9** [comment]
Hello @voodoos and thank you for your response.

> what your are trying to achieve is not entirely clear to me. Are you using odoc for comparison with Merlin results or do you plan to mix the two ?

I'm trying to mix the two to build and extra layer to perform an analysis on comments, see [ometrics](https://gitlab.com/nomadic-labs/ometrics). I'm trying to build a list of entry (module, types, values etc.) and determine whether they are commented or not.

> Also I think there is indeed an issue with Merlin but it is not linked to the licence header: can you try removing the space between the documentation string and the type declaration ?

My example was poorly described, I did no mean the documentation of type `t` but the whole module documentation:
```ocaml
(*******************************)
(* This is a wonderful license *)
(*******************************)

(** Toplevel documentation *)

(* any OCaml code *)
```
If I run my executable written in this "minimal" [repo](https://github.com/vch9/merlin-analysis-module) on a file `example.ml` with the content given above I get:
```
{ kind : Module; name : Example; documented : false }
```
Meaning: I did not achieve to associate the comment `(** Toplevel documentation *)` to the module `Example`.

Regarding merlin, If I try `merlin-document` on the module `Example` when the license header is present I get `(*No documentation available)` whereas If I remove the license header I get: `(*Toplevel documentation)`
